profiler/src/ProfilerEngine/Datadog.Profiler.Native/FrameStore.cpp

```diff
@@ -896,6 +928,182 @@ std::pair<std::string, mdTypeDef> FrameStore::GetMethodNameFromMetadata(IMetaDat
+    // - fetch each parameter properties to get its name
+    // NOTE: in case of non static method, the implicit 'this' parameter is not part of the parameters
+    //       neither in the signature nor in the metadata enumeration EnumParams
+    hr = S_OK;
+    HCORENUM hEnum = 0;
+    ULONG paramCount;
+
+    // Note: calling EnumParams with NULL, 0 to get the size does not seem to work
+    //       so rely on the count provided by the blob
+    auto paramDefs = std::make_unique<mdParamDef[]>(argCount);
+    hr = pMetaData->EnumParams(&hEnum, mdTokenFunc, paramDefs.get(), argCount, &paramCount);
+    pMetaData->CloseEnum(hEnum);
+
+    // sanity checks but one time, paramCount was greater than argCount so remove it
+    //assert(paramCount == argCount);
+
+    ULONG pos;
+    WCHAR name[260];
+    ULONG length;
+
+    DWORD bIsValueType;
+    ULONG currentGenericParam = 0;
+    builder << "(";
+    for (ULONG i = 0;
+         (SUCCEEDED(hr) && (pSigBlob != NULL) && (i < (argCount)));
+         i++)
+    {
+        // get the parameter name
+        hr = pMetaData->GetParamProps(paramDefs[i], NULL, &pos, name, ARRAY_LEN(name) - 1, &length, &attributes, &bIsValueType, NULL, NULL);
+        // note that we need to convert from WCHAR* to char* for the name
+
+        // get the parameter type
+
+        // In case of generic function, get the type details from the runtime and not from the metadata
+        // We don't know in advance which parameter is a generic parameter and this is given by elementType == MVAR
+        mdToken parameterTypeToken = mdTypeDefNil;
+        std::stringstream parameterBuilder;
+        pSigBlob = ParseElementType(pMetaData, pSigBlob, classTypeArgs.get(), methodTypeArgs.get(), &elementType, parameterBuilder, &parameterTypeToken);
+        if ((methodTypeArgs != NULL) && (elementType == ELEMENT_TYPE_MVAR))
+        {
+            // the offset in the array is at the end of the type name read from the metadata
+            // ex: M7 --> offset 7
+            int offset = std::stoi(parameterBuilder.str().substr(1));
+            if (offset < genericArgCount)
+            {
+                std::string sTypeName;
+                if (GetTypeName(methodTypeArgs[offset], sTypeName))
+                {
+                    builder << sTypeName;
+                    builder << " ";
+                    // convert from UTF16 to UTF8
+                    builder << shared::ToString(shared::WSTRING(name));
+                }
+                else
+                {
+                    builder << "?";
+                }
+            }
+            else
+            {
+                // case detected during tests with System.MemoryExtensions.IndexOfAny<T>(ReadOnlySpan<T>, T, T) for example
+                // but should be treated now
+                builder << "M" << offset << "?";
+                // convert from UTF16 to UTF8
+                builder << shared::ToString(shared::WSTRING(name));
```

Comment:
here you may use the `shared::ToString(..)` overload for `WCHAR`

Reply:
✅ done